gsy_framework/data_classes.py

```diff
@@ -295,44 +264,10 @@ def serializable_dict(self) -> Dict:
         """Return a json serializable representation of the class."""
         return {**super().serializable_dict(), "buyer": self.buyer.serializable_dict()}
 
-    @classmethod
-    def from_json(cls, offer_or_bid: str) -> "Bid":
-        bid = super().from_json(offer_or_bid)
-        json_dict = json.loads(offer_or_bid)
-        if "buyer" in json_dict:
-            bid.buyer = TraderDetails(
-                name=json_dict["buyer"]["name"],
-                uuid=json_dict["buyer"]["uuid"],
-                origin=json_dict["buyer"]["origin"],
-                origin_uuid=json_dict["buyer"]["origin_uuid"],
-            )
-        return bid
-
-    def to_json_string(self, **kwargs) -> str:
-        """Convert the Offer object into its JSON representation."""
-        if self.buyer:
-            kwargs["buyer"] = self.buyer.serializable_dict()
-        return super().to_json_string(**kwargs)
-
     @staticmethod
-    def from_dict(bid: Dict) -> "Bid":
-        """Deserialize a bid dict."""
-
-        return Bid(
-            id=bid["id"],
-            creation_time=str_to_pendulum_datetime(bid["creation_time"]),
-            energy=bid["energy"],
-            price=bid["energy"] * bid["energy_rate"],
-            original_price=bid.get("original_price"),
-            buyer=TraderDetails(
-                name=bid["buyer"]["name"],
-                origin=bid["buyer"]["origin"],
-                origin_uuid=bid["buyer"]["origin_uuid"],
-                uuid=bid["buyer"]["uuid"]),
-            attributes=bid.get("attributes"),
-            requirements=bid.get("requirements"),
-            time_slot=str_to_pendulum_datetime(bid.get("time_slot"))
-        )
+    def from_dict(offer: Dict) -> "Offer":
+        """Deserialize an offer dict."""
+        return Offer.from_serializable_dict(offer)
```

Comment:
Indeed, and this was working as expected till now and not caught by the tests because the actual creation of the Bid / Offer object was in the base class, therefore incidentally working correctly! I removed the implementation from both Bid and Offer class and moved it to the BaseBidOffer class. Thanks!

Reply:
Thank you very much. Is any unit test needed to avoid this in the future?

Reply:
Yes, there are 2 tests for the Bid and Offer respectively, both named `test_from_dict` , where the method return value is validated.